great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -1740,9 +1740,7 @@ def _instantiate_datasource_from_config_and_update_project_config(
             DatasourceInitializationError
         """
         if save_changes:
-            self._datasource_store.set_by_name(  # type: ignore[attr-defined]
-                datasource_name=config.name, datasource_config=config
-            )
+            config = self._datasource_store.set(key=None, value=config)  # type: ignore[attr-defined]
```

Comment:
Now that `set()` also calls `get()` to retrieve and return the newly created config (which will be populated with IDs in the cloud store case) we use that config in the datasource cache and to instantiate the datasource.